lib/checkDupHostingKeys.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+var _ = require('lodash');
+var RSVP = require('rsvp');
+var utils = require('./utils');
+var chalk = require('chalk');
+var Config = require('./config');
+var FirebaseError = require('./error');
+var logger = require('./logger');
+
+module.exports = function(options) {
+  return new RSVP.Promise(function(resolve, reject) {
+    if (!options.config) {
+      return reject(new FirebaseError('Missing firebase.json.'));
+    }
+
+    var src = options.config._src;
+    var legacyKeys = Config.LEGACY_HOSTING_KEYS;
+
+    var hasLegacyKeys = _.reduce(legacyKeys, function(result, key) {
+      return result || _.has(src, key);
+    }, false);
+
+
+    if (hasLegacyKeys && _.has(src, ['hosting'])) {
+      utils.logWarning(chalk.bold.yellow('[hosting] ') + 'We found a ' + chalk.bold('hosting') + ' key inside ' + chalk.bold('firebase.json') + ' as well as hosting configuration keys that are not nested inside the ' + chalk.bold('hosting') + ' key.');
```

Comment:
Let's make this consistent with deploy bullets, so `hosting:` in yellow instead of `[hosting]`